File: Problems/0400_0499/0461_Hamming_Distance/Project_CS/Hamming_Distance.cs
using System;
using System.Linq;

public class Solution
{
    public int HammingDistance(int x, int y)
    {
        int z = x ^ y;
        int counts = 0;
        for ( ;z > 0; z /= 2)
        {
            if (z % 2 == 1)
            {
                counts++;
            }
        }

        return counts;
    }

    public int HammingDistance2(int x, int y)
    {
        int xor = x ^ y;
        int distance = 0;
        while (xor > 0)
        {
            distance += xor & 1;
            xor = xor >> 1;
        }
        return distance;
    }

    public void Main(string args)
    {
        string[] flds = args.Replace("[","").Replace("]","").Trim().Split(',');
        int x = int.Parse(flds[0]);
        int y = int.Parse(flds[1]);

        Console.WriteLine("x = " + x.ToString() + ", y = " + y.ToString());
        
        System.Diagnostics.Stopwatch sw = new System.Diagnostics.Stopwatch();
        sw.Start();

        int result = HammingDistance(x, y);
        Console.WriteLine("result = " + result.ToString());
        
        sw.Stop();
        Console.WriteLine("Execute time ... " + sw.ElapsedMilliseconds.ToString() + "ms\n");
    }
}

File: Problems/0400_0499/0461_Hamming_Distance/Project_Python3/Hamming_Distance.py
# coding: utf-8

import os
import sys
import time

class Solution:
    def hammingDistance(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        return bin(x ^ y).count('1')

    def hammingDistance_work(self, x, y):
        """
        :type x: int
        :type y: int
        :rtype: int
        """
        counts = 0
        while x > 0 or y > 0:
            if x % 2 != y % 2:
                counts += 1
            x //= 2
            y //= 2
        return counts

def main():
    argv = sys.argv
    argc = len(argv)

    if argc < 2:
        print("Usage: python {0} <testdata.txt>".format(argv[0]))
        exit(0)

    if not os.path.exists(argv[1]):
        print("{0} not found...".format(argv[1]))
        exit(0)

    testDataFile = open(argv[1], "r")
    lines = testDataFile.readlines()

    for temp in lines:
        temp = temp.strip()
        if temp == "":
            continue
        print("args = {0}".format(temp))
        loop_main(temp)
    #   print("Hit Return to continue...")
    #   input()

def loop_main(temp):
    flds = temp.replace("[","").replace("]","").rstrip().split(",")
    x = int(flds[0])
    y = int(flds[1])

    sl = Solution()
    time0 = time.time()
    result = sl.hammingDistance(x, y)

    print("result = {0}".format(result))

    time1 = time.time()
    print("Execute time ... : {0:f}[s]\n".format(time1 - time0))

if __name__ == "__main__":
    main()
